### PR TITLE
Roman Numeral Conversion in C#

### LINKED ISSUE #592: Add Roman Numeral Conversion in C#

### PR DESCRIPTION
**Pull request by auroq**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [x] I fixed #592 

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

File: archive/c/c-sharp/roman-numeral.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        };
+
+        private static int RomanToDecimal(string roman, int total=0)
+        {
+            var romanArray = roman.ToCharArray();
+            if (romanArray.Length == 1)
+                return total + RomanDecMapping[romanArray[0]];
+
+            var romanVal = RomanDecMapping[romanArray[0]];
+            var nextRomanVal = RomanDecMapping[romanArray[1]];
```
Comment (chrboe) [inline archive/c/c-sharp/roman-numeral.cs]: I'll need some opinions from @TheRenegadeCoder/core on this, but I'm pretty sure this causes an exception if the character is not a roman digit. Do we want to introduce a sanity check here, albeit at a expense of brevity?

Comment (jrg94) [reply archive/c/c-sharp/roman-numeral.cs]: I'm glad you pointed this out. I had to revisit the [original solution which was in Python](https://github.com/TheRenegadeCoder/sample-programs/blob/master/archive/p/python/roman-numeral.py). After digging into it a bit, I noticed that for invalid symbols, we return -1. However, we never actually verify that the input is valid.

As a result, input like `XT` will result in -9 which isn't true. In fact, this is an error, and I think it should be addressed in both solutions as well as updated in the rules.

How we address it though may prove to be too complicated though, so let's see what we can come up with.

Comment (jrg94) [reply archive/c/c-sharp/roman-numeral.cs]: Hey, I just updated the roman numeral docs. Let me know how you like them for addressing this issue. #595

**Comment by auroq** [comment]
@jrg94 I can certainly add some handling for invalid input, but how far would you like me to go. My knee jerk reaction would be to use regex to verify it's a valid Roman numeral, but that could add some complexity that I think this project is trying to avoid.

Would it be sufficient to just verify that all inputted characters are in the set of Roman numerals? 

That would prevent the user from inputting "XT" , but would not prevent inputting something like "XID" which contains all valid numerals but is not valid altogether.

**Comment by auroq** [comment]
This got closed somewhere along the way in the session issues with github. I'm reopening it now, and I think my two commits should both be there now. I just surround the function with a try catch and output the error as you had it in the new documentation. Let me know if that's what you're looking for.

**Comment by auroq** [comment]
This got closed somewhere along the way in the session issues with github. I'm reopening it now, and I think my two commits should both be there now. I just surround the function with a try catch and output the error as you had it in the new documentation. Let me know if that's what you're looking for.

**Comment by auroq** [comment]
This got closed somewhere along the way in the session issues with github. I'm reopening it now, and I think my two commits should both be there now. I just surround the function with a try catch and output the error as you had it in the new documentation. Let me know if that's what you're looking for.

**Comment by jrg94** [comment]
I'm still having session issues.

**Comment by auroq** [comment]
This got closed somewhere along the way in the session issues with github. I'm reopening it now, and I think my two commits should both be there now. I just surround the function with a try catch and output the error as you had it in the new documentation. Let me know if that's what you're looking for.

**Comment by auroq** [comment]
There's an issue with session storage on github right now. I've already
updated my changes, but can't push them until github comes back. I'm
watching their status pageand will push when it's up.

**Pull request merged**